config/initializers/101-lograge.rb

```diff
@@ -1,10 +1,47 @@
 # frozen_string_literal: true
 
-Rails.application.config.to_prepare do
-  if (Rails.env.production? && SiteSetting.logging_provider == "lograge") ||
-       (ENV["ENABLE_LOGRAGE"] == "1")
-    require "lograge"
+if (Rails.env.production? && SiteSetting.logging_provider == "lograge") ||
+     (ENV["ENABLE_LOGRAGE"] == "1")
+  require "lograge"
+
+  Rails.application.config.after_initialize do
+    def unsubscribe(component_name, subscriber)
+      subscriber
+        .public_methods(false)
+        .reject { |method| method.to_s == "call" }
+        .each do |event|
+          ActiveSupport::Notifications
+            .notifier
+            .all_listeners_for("#{event}.#{component_name}")
+            .each do |listener|
+              if listener
+                   .instance_variable_get("@delegate")
+                   .class
+                   .to_s
+                   .start_with?("#{component_name.to_s.classify}::LogSubscriber")
```

Comment:
Currently, the `lograge` gem is doing `if listener.instance_variable_get('@delegate') == subscriber`. However since Rails 7.1, `@delegate` can be [an instance of `ActionView::LogSubscriber::Start`](https://github.com/rails/rails/blob/16f1222753cac0a554a2acd8f4ffa3f23d51d945/actionview/lib/action_view/log_subscriber.rb#L106-L107) instead of always being an instance of `ActionView::LogSubscriber`.